metadata/metadata.js

```diff
@@ -7,70 +7,113 @@
- * @param {String} namespace name of the namespace
- * @returns {String|null} the metadata as a string, or null
+ * @memberOf metadata
  */
-let getValue = function(itemName, namespace) {
-    let result = MetadataRegistry.get(new MetadataKey(namespace, itemName));
-    return result ? result.value : null;
-};
+class ItemMetadata {
 
-let addValue = function(itemName, namespace, value) {
-    let key = new MetadataKey(namespace, itemName);
-    MetadataRegistry.add(new Metadata(key, value, {}));
-}
+    /**
+     * Don't use this constructor, instead call {@link items.Item.metadata}.
+     * @param {String} itemName
+     * @hideconstructor
+     */
+    constructor(itemName) {
+        this.itemName = itemName;
+    }
 
-let updateValue = function(itemName, namespace, value) {
-    let metadata = createMetadata(itemName, namespace, value);
-    let result = MetadataRegistry.update(metadata);
-    return result ? result.value : null;
-}
+    /**
+     * This function will return the Metadata value associated with the
+     * specified Item.
+     * 
+     * @param {String} namespace name of the namespace
+     * @returns {String|null} the metadata value as a string, or null
+     */
+    getValue(namespace) {
+        let result = MetadataRegistry.get(new MetadataKey(namespace, this.itemName));
+        return result ? result.value : null;
+    };
 
-/**
- * Adds (inserts) or updates a metadata value.
- * 
- * @param {String} itemName the name of the item
- * @param {String} namespace the name of the namespace
- * @param {String} value the value to insert or update
- * @returns {Boolean} true if the value was added, false if it was updated
- */
-let upsertValue = function(itemName, namespace, value) {
-    let existing = getValue(itemName, namespace);
+    /**
+    * Adds (inserts) a metadata value.
+    * 
+    * @param {String} namespace the name of the namespace
+    * @param {String} value the value to insert or update
+    */
+    addValue(namespace, value) {
+        let key = new MetadataKey(namespace, this.itemName);
+        MetadataRegistry.add(new Metadata(key, value, {}));
+    }
 
-    if (existing === null) {
-        addValue(itemName, namespace, value);
-        return true;
-    } else {
-        updateValue(itemName, namespace, value);
-        return false;
+    /**
+    * Updates metadata values for this item.
+    * @param {String} namespace The namespace for the metadata to update
+    * @param {String} value the value to update the metadata to
+    * @returns {String} the updated value
+    */
+    updateValue(namespace, value) {
+        let metadata = createMetadata(this.itemName, namespace, value);
+        let result = MetadataRegistry.update(metadata);
+        return result ? result.value : null;
+    }
+
+    /**
+     * Updates metadata values for this item.
+     * @param {Map} namespaceToValues A map of namespaces to values to update
+     */
+    updateValues(namespaceToValues) {
```

Comment:
Here the `namespaceToValues` map should support configuration too. But it's worth to document how to prepare that structure. Wdyt ?